### PR TITLE
Tidy up Contributors Docs

### PR DESCRIPTION
**Pull request by gundalow**
##### SUMMARY
* Thin down `README` so it's not a wall of text
* Remove duplicated info
* Direct people to docs.ansible.com rather than repeating information (badly) in multiple palces
* Move information to where it should belong

##### ISSUE TYPE
- Docs Pull Request

### CODE REVIEW

File: CODING_GUIDELINES.md [hunk 1]
```diff
@@ -1,316 +1,10 @@
...
-  * Four space indent is strictly required
-  * Include meaningful whitespace between lines of code
-
-Shebang Lines
-=============
-
-  * /usr/bin/scripts should start with '/usr/bin/env python'
-  * module code should still use '/usr/bin/python' as this is replaced automatically by 'ansible_python_interpreter', see the FAQ in the docs for more info.
-
-Comments
```
Comment (gundalow) [inline CODING_GUIDELINES.md:49]: Maybe port this to dev_guide?

File: CODING_GUIDELINES.md [hunk 2]
```diff
@@ -1,316 +1,10 @@
...
-========
-
-  * Readability is one of the most important goals for this project
-  * Comment any non-trivial code where someone might not know why you are doing something in a particular way
-  * Though if something should be commented, that's often a sign someone should write a function
-  * All new functions must have a basic docstring comment
-  * Commenting above a line is preferable to commenting at the end of a line
-
-Classes
```
Comment (gundalow) [inline CODING_GUIDELINES.md:58]: Maybe port this to dev_guide?

File: CODING_GUIDELINES.md [hunk 3]
```diff
@@ -1,316 +1,10 @@
...
-  * "Don't repeat yourself" is generally a good philosophy
-
-Variables
-=========
-
-  * Use descriptive variable names instead of variables like 'x', unless x is an obvious loop index
-  * Ansible python code uses identifiers like 'ClassesLikeThis and variables_like_this
-  * Module parameters should also use_underscores and not runtogether
-
-Module Security
```
Comment (gundalow) [inline CODING_GUIDELINES.md]: Maybe port this to dev_guide?

**Review by acozine — APPROVED** [review]
This is a great set of changes -it's so good to see use DRYing out the docs. 

I reorganized the material added to the dev_guide a bit - with that change a big +1 from me.

**Pull request merged**